### PR TITLE
Don't do anything when not logged into GitHub

### PR DESCRIPTION
**Pull request by ramlmn**
Perform specific actions _only_ when user is logged in.

### CODE REVIEW

**Comment by sindresorhus** [comment]
I think we should just not do anything when the user is logged out. This extension is for users logged into GitHub. I don't want conditionals all over the codebase for this. Better to just not activate any of the code when not logged in.

**Comment by ramlmn** [comment]
I think you can still do some things even when the user is logged out like linkify urls, show real names and reaction participants, copy markdown, and add copy buttons.

I don't know if other people use this plugin when logged out but, I almost always browse other people codebases in _incognito_ mode.

But still open to know other peoples opinion on this PR.

**Comment by bfred-it** [comment]
I agree with Sindre, the problem with this direction is that none of us uses it without logging in so we'll always end up breaking something or not supporting it fully. Either we leave it as is (and whatever works works) or we add a `if !username return` in `init`

File: src/content.js [hunk 1]
```diff
@@ -532,10 +537,15 @@ async function onDomReady() {
 	await domLoaded;
 
 	const username = getUsername();
+	if (!username) {
+		return;
+	}
```
Comment (bfred-it) [inline src/content.js]: No need for this; onDomReady is called by init

Comment (ramlmn) [inline src/content.js]: Sorry, though it was being called for each page load.

**Comment by sindresorhus** [comment]
Thanks @ramlmn. Keep 'em coming :)

**Comment by ramlmn** [comment]
I'm glad to be a part of the community 😋.

**Comment by sindresorhus** [comment]
Actually, the username checks throws an error as it tries to get the DOM element before the DOM is ready. [I've disabled it for now.](https://github.com/sindresorhus/refined-github/commit/12a321ad11b4c065f66ffe066af6502522667253)

```
Uncaught TypeError: Cannot read property 'getAttribute' of null
    at exports.getUsername (content.js:551)
    at init (content.js:1519)
    at Object.<anonymous> (content.js:1690)
    at __webpack_require__ (content.js:20)
    at content.js:63
    at content.js:66
```

**Comment by ramlmn** [comment]
@sindresorhus Just move line [`L498`](https://github.com/sindresorhus/refined-github/blob/master/src/content.js#L498) to [`L439`](https://github.com/sindresorhus/refined-github/blob/master/src/content.js#L439).

**Comment by sindresorhus** [comment]
But then we are deferring a lot of things that doesn't need to wait on DOM ready. I don't think that's the best solution.

**Comment by ramlmn** [comment]
That's the best one I can think of.

Or you could `await elementReady('meta[name="user-login"]')` in `getUsername`.

**Pull request merged**